django/contrib/postgres/fields/uuid.py

```diff
@@ -0,0 +1,21 @@
+from django.contrib.postgres.functions import RandomUUID
+from django.db.models import UUIDField
+from django.db.models.fields import NOT_PROVIDED, AutoFieldMixin
+
+
+class UUID4Field(UUIDField):
+    def __init__(self, *args, **kwargs):
+        kwargs["db_default"] = RandomUUID()
```

Comment:
Why we only define it on PostgreSQL? It should be possible to define it on other dbs, e.g. MySQL.

Reply:
For me the goal of this ticket/PR is to allow to define custom auto primary keys, such as UUIDs, generated in the DB, used in tandem with `DEFAULT_AUTO_FIELD`. This PR allows to do just that and uuid4 auto for psql just proves that it works.

If there was an already existing `RandomUUID` equivalent for MySQL, I'd be happy to use it, but there's none (in the latest Django version), and implementing it is IMO out of scope for this PR.

Let me know what you think.

Reply:
It seems like a good approach to me.